app/controllers/projects_controller.rb

```diff
@@ -30,10 +30,24 @@ def send_project_csv
     send_data project_csv, filename: project_name + '_sample-table.csv'
   end
 
-  def project_reports_csv
-    output_file = bulk_report_csvs_from_params(@project, params)
-    return unless output_file
-    send_file output_file
+  def make_project_reports_csv
+    `aws s3 rm #{@project.report_tar_s3}`
+    Resque.enqueue(GenerateProjectReportsCsv, params)
```

Comment:
we need to make sure the output file is id'ed by user because each user might see different content once we put in the access control.

Reply:
Ok!

Reply:
Good catch, @yunfangjuan.   I think it would be slightly more optimal if id'ed on visibility rather than user.   I.e. if the user's access is "public" then they can get the "report_public.tar", and if the user's access is "contributor" then they can get the full "report.tar".   That way when a public link to a report is shared, the public doesn't overwhel our servers with CSV generation -- but instead each public member immediately downloads the existing "report_public.tar".